src/V3LinkDot.cpp

```diff
@@ -1961,8 +1961,18 @@ class LinkDotResolveVisitor : public AstNVisitor {
             m_ds.m_dotp = nodep;  // Always, not just at start
             m_ds.m_dotPos = DP_SCOPE;
 
-            // m_ds.m_dotText communicates the cell prefix between stages
-            if (VN_IS(nodep->lhsp(), ClassOrPackageRef)) {
+            if (VN_IS(nodep->lhsp(), ParseRef) && nodep->lhsp()->name() == "this") {
+                VSymEnt* classSymbolTable = m_ds.m_dotSymp;
+                do {
+                    classSymbolTable = classSymbolTable->fallbackp();
+                } while (classSymbolTable && !VN_IS(classSymbolTable->nodep(), Class));
+                m_ds.m_dotSymp = classSymbolTable;
+                if(!classSymbolTable) {
```

Comment:
Please run "make clang-format" to standardize indentation.  If your version of clang is recent this will only change your stuff, if it changes a lot of other stuff don't worry about it but put a space before the ( here.

Reply:
Fixed.